### PR TITLE
modularize and update code

### LINKED ISSUE #18: Listing shouldn't require Github description
Right now to list a package we require them to have a Github repo description.

This requirement looks unnecessary and should be removed.

### PR DESCRIPTION
**Pull request by eddiemonge**
* Style changes
 * Break up the index file into component pieces
 * Optimize some code
 * Update output of api call
 * add more logging

Fixes #18 

Was thinking https://github.com/trentm/node-bunyan might be good for logging and can be dropped in easily enough

### CODE REVIEW

File: plugin-cache/index.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+'use strict';
+
+var updateList = require('./src/update');
+
+var log = process.env.LOGGER || console;
+
+module.exports = function (keyword, limit) {
+  if (typeof keyword === 'undefined') {
```
Comment (sindresorhus) [inline plugin-cache/index.js]: `if (typeof keyword !== 'string') {`

File: plugin-cache/src/gh-stats.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+var log = process.env.LOGGER || console;
+
+module.exports = function (list, limit) {
+  limit = limit || list.length;
+  limit = list.length < limit ? list.length : limit;
+  log.info('Fetching GH stats for %s packages', limit);
+
+  var count = 0;
+  var limitMessage = false;
+  var re = /github\.com\/([\w\-\.]+)\/([\w\-\.]+)/i;
```
Comment (sindresorhus) [inline plugin-cache/src/gh-stats.js]: This could use https://github.com/zeke/github-url-to-object instead.

Comment (eddiemonge) [inline plugin-cache/src/gh-stats.js]: added it

**Comment by eddiemonge** [comment]
Anything else need to be changed?

File: plugin-cache/src/npm-download-counts.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
+'use strict';
+
+var http = require('https');
+
+var Q = require('q');
+
+var packageJson = require('package-json');
+
```
Comment (kevva) [inline plugin-cache/src/npm-download-counts.js]: Why is there an empty row between all `require`?

File: plugin-cache/src/merge-info.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    if (npm.description.length > gh.description.length) {
+      description = npm.description;
+    } else {
+      description = gh.description;
+    }
+  } else {
+    description = npm.description || gh.description || '';
+  }
+
+  var owner = npm.author && npm.author.name
```
Comment (kevva) [inline plugin-cache/src/merge-info.js]: `var owner = npm.author && npm.author.name || gh.owner && gh.owner.login || '';`

File: plugin-cache/src/merge-info.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  owner = owner || (gh.owner && gh.owner.login) || '';
+
+  var ownerWebsite = npm.author && npm.author.url
```
Comment (kevva) [inline plugin-cache/src/merge-info.js]: `var ownerWebsite = npm.author && npm.author.url|| gh.owner && gh.owner.html_url || '';`

File: plugin-cache/src/merge-info.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+  ownerWebsite = ownerWebsite || (gh.owner && gh.owner.html_url) || '';
+
+  return {
+    name: npm.name,
+    description: description,
+    stars: gh.stargazers_count || 0,
+    downloads: npm.downloads,
+    website: npm.website || gh.html_url || '',
+    owner: owner,
```
Comment (kevva) [inline plugin-cache/src/merge-info.js]: `owner: owner || '',` (if we need the empty strings)

File: plugin-cache/src/merge-info.js [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    ownerWebsite: ownerWebsite,
```
Comment (kevva) [inline plugin-cache/src/merge-info.js]: `ownerWebsite: ownerWebsite || '',`

File: plugin-cache/src/npm-download-counts.js (hunk 3 above)
Comment (eddiemonge) [inline plugin-cache/src/npm-download-counts.js]: I don't know. I logically separate them in my head like that, node specific, 3rd party helpers, actual dependencies. No other reason

File: plugin-cache/src/merge-info.js (hunk 4 above)
Comment (eddiemonge) [inline plugin-cache/src/merge-info.js]: thats pretty much how it is with an additional fallback of an empty string

**Comment by eddiemonge** [comment]
Anything else?

**Comment by SBoudrias** [comment]
Looks good enough for me. Let's merge!

**Pull request merged**